envoyauth/response.go

```diff
@@ -307,7 +307,7 @@ func (result *EvalResult) GetDynamicMetadata() (*_structpb.Struct, error) {
 
 		metadata, ok := val.(map[string]interface{})
 		if !ok {
-			return nil, fmt.Errorf("type assertion error")
+			return nil, fmt.Errorf("type assertion error, dynamic_metadata is not a map[string]interface{}")
```

Comment:
`map[string]interface{}` is a Go type, not a Rego one. I'd just say "is not an object" here. Same for headers below.